patches/server/0947-Player-Entity-Tracking-Events.patch

```diff
@@ -0,0 +1,43 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Yannick Lamprecht <yannicklamprecht@live.de>
+Date: Wed, 30 Mar 2022 18:16:52 +0200
+Subject: [PATCH] Player Entity Tracking Events
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
+index 6fa47becd0f83ac4273ef3a10c314aa27b08184b..200cc4e3e75992ee03f24e3ee78640a809a9217d 100644
+--- a/src/main/java/net/minecraft/world/entity/Entity.java
++++ b/src/main/java/net/minecraft/world/entity/Entity.java
+@@ -7,6 +7,8 @@ import com.google.common.collect.Lists;
+ import com.google.common.collect.Sets;
+ import com.google.common.collect.UnmodifiableIterator;
+ import com.mojang.logging.LogUtils;
++import io.papermc.paper.event.player.PlayerTrackEntityEvent;
++import io.papermc.paper.event.player.PlayerUntrackEntityEvent;
+ import it.unimi.dsi.fastutil.objects.Object2DoubleArrayMap;
+ import it.unimi.dsi.fastutil.objects.Object2DoubleMap;
+ import java.util.Arrays;
+@@ -3796,9 +3798,21 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource {
+         EnchantmentHelper.doPostDamageEffects(attacker, target);
+     }
+ 
+-    public void startSeenByPlayer(ServerPlayer player) {}
++    public void startSeenByPlayer(ServerPlayer player) {
++        // Paper start
++        if (PlayerTrackEntityEvent.getHandlerList().getRegisteredListeners().length == 0 ||
++            new PlayerTrackEntityEvent(player.getBukkitEntity(), this.getBukkitEntity()).callEvent()) {
++        }
++        // Paper end
++    }
+ 
+-    public void stopSeenByPlayer(ServerPlayer player) {}
++    public void stopSeenByPlayer(ServerPlayer player) {
++        // Paper start
++        if(PlayerUntrackEntityEvent.getHandlerList().getRegisteredListeners().length > 0) {
++            new PlayerUntrackEntityEvent(player.getBukkitEntity(), this.getId(), this.getBukkitEntity()).callEvent();
++        }
++        // Paper end
++    }
```

Comment:
Wrap the whole method in paper comments as technically you're moving the brackets